org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingRecordPatternTest.java

```diff
@@ -585,4 +585,63 @@ public void testModifyRecordInstanceOfPattern() throws Exception {
 		assertEqualString(preview, buf.toString());
 	}
 
+	public void testNPEinASTRewriteFlattener() throws Exception {
```

Comment:
ASTRewritingSwitchPatternTest is the right place for this test as there is no RecordPattern in the test case. Please move this test there

Reply:
the test case moved to **ASTRewritingSwitchPatternTest**.